File: python/src/app/scheduling/shortest_job_first_scheduler.py
from python.src.app.scheduling.job import Job
from python.src.app.scheduling.scheduler import Scheduler
from python.src.util.minheap import MinHeap


class ShortestJobFirstScheduler(Scheduler):

    def __init__(self, num_jobs):
        self.jobs = MinHeap()

        for i in range(num_jobs):
            self.jobs.insert(Job())

    def simulate(self):
        # TODO: complete this function
        return 0

File: java/src/edu/berkeley/cs/app/scheduling/ShortestJobFirstScheduler.java
package edu.berkeley.cs.app.scheduling;

import edu.berkeley.cs.util.MinHeap;

public class ShortestJobFirstScheduler implements Scheduler {
  protected MinHeap<Job> jobs;

  public ShortestJobFirstScheduler(int numJobs) {
    jobs = new MinHeap<>();

    for (int i = 0; i < numJobs; i++) {
      jobs.insert(new Job());
    }
  }

  /**
   * Determine the time units required to run the shortest job first scheduling algorithm. In this
   * algorithm, jobs are given an unlimited amount of time to process (e.g. use Integer.MAX_VALUE).
   *
   * <p>Remember to add the time a given job took to those jobs still waiting in the queue in order
   * to simulate them waiting for processor time.
   */
  @Override
  public int simulate() {
    // TODO: complete this function
    return 0;
  }
}
